### PR TITLE
Added files validation

### PR DESCRIPTION
**Pull request by rvyuha**
The focus of this PR is the initial files sheet validation.
The following changes can be found:

- files-validation.R contains the main function used for the dictionary automation.
- files.R contains the hardcoded strings used to interface with the files sheet.
- tests/ Contains test files.
- data/test-data/ Contains test excel files.
- .github/workflows/update-documentation.yml new github action workflow to trigger the automation workflow.

### CODE REVIEW

**Comment by rvyuha** [comment]
@yulric ready for review

File: R/files-validation.R [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+  # Remove any rows with not supported fileType.
+  files_sheet_formatted <-
+    files_sheet[files_sheet$fileType %in% files$file_type$categories, unlist(files_sheet_column_names)]
+  # insert version
+  files_sheet_formatted$name <-
+    gsub('\\{version\\}', version, files_sheet_formatted$name)
+  csv_to_extract <- c()
+  excel_to_extract <- list()
+  errors <- ""
+  for (row_index in 1:nrow(files_sheet_formatted)) {
```
Comment (yulric) [inline R/files-validation.R:118]: Use `seq_len` in case there are 0 rows?

File: R/files-validation.R [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    working_row <- files_sheet_formatted[row_index,]
+    partID <- working_row[[files$part_ID$name]]
+    fileType <- working_row[[files$file_type$name]]
+    if (fileType == files$file_type$categories$csv) {
+      sets_info <- sets_sheet[sets_sheet$setID == partID, "partID"]
```
Comment (yulric) [inline R/files-validation.R:123]: Why are you pulling out the `partID` column here?

File: R/files-validation.R [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+      if (nrow(sets_info) >= 1) {
+        # Append error
+        errors <-
+          paste0(errors,
```
Comment (yulric) [inline R/files-validation.R:127]: Can we add more information to the errors? For example the `fileID`? To make it easy for the user to know where the error is.

File: R/files-validation.R [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+  files_sheet_formatted$name <-
+    gsub('\\{version\\}', version, files_sheet_formatted$name)
+  csv_to_extract <- c()
+  excel_to_extract <- list()
+  errors <- ""
+  for (row_index in 1:nrow(files_sheet_formatted)) {
+    working_row <- files_sheet_formatted[row_index,]
+    partID <- working_row[[files$part_ID$name]]
+    fileType <- working_row[[files$file_type$name]]
+    if (fileType == files$file_type$categories$csv) {
```
Comment (yulric) [inline R/files-validation.R:122]: Shouldn't you also check if it is a part, that it exists in the part sheet?

File: R/files-validation.R [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+      } else{
+        csv_to_extract <- c(csv_to_extract, partID)
+      }
+    } else if (fileType == files$file_type$categories$excel) {
+      sets_info <- sets_sheet[sets_sheet$setID == partID, "partID"]
+      if (nrow(sets_info) >= 1) {
+        excel_to_extract[[partID]] <- sets_info
+      } else{
+        # Append error
+        errors <-
```
Comment (yulric) [inline R/files-validation.R:140]: The `partID` column for an excel sheet can also be a part, it doesn't have to be a set. https://github.com/Big-Life-Lab/PHES-ODM/blob/release-automation/specs/release-automation/release-automation-specs.qmd#L230

File: R/files.R [hunk 6]
```diff
@@ -0,0 +1,33 @@
+files <- list(
```
Comment (yulric) [inline R/files.R:1]: Rename to `files_sheet`

File: R/files.R [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    name = "addHeaders"
+  ),
+  destinations = list(
+    name = "destinations",
+    categories = list(
+      osf = "osf",
+      github = "github"
+    )
+  ),
+  osf_locations = list(
```
Comment (yulric) [inline R/files.R:25]: osf_locations -> osf_location

File: R/odm-dictionary-file.R [hunk 8]
```diff
@@ -0,0 +1,9 @@
+odm_dictionary <- list()
+
+odm_dictionary$dictionary_directory <- "data/raw"
```
Comment (yulric) [inline R/odm-dictionary-file.R:3]: Aren't the first two variables (dictionary_directory and tmp_dictionary_directory) not applicable to an ODM dictionary but only to the automation? If so, it should be moved to the files-validation.R file.

File: R/odm-dictionary-file.R [hunk 9]
```diff
@@ -0,0 +1,9 @@
...
+odm_dictionary$tmp_dictionary_directory <- "data/tmp"
+odm_dictionary$dictionary_missing_value_replacement <- "NA"
```
Comment (yulric) [inline R/odm-dictionary-file.R:5]: Is this used?

File: .github/update-documentation.yml [hunk 10]
```diff
@@ -0,0 +1,26 @@
+name: Update Documentation
```
Comment (yulric) [inline .github/update-documentation.yml:1]: Not sure Update Documentation makes sense as the name. Maybe something like `Create Release`

**Review by yulric — CHANGES_REQUESTED** [review]
* Can we move all the files for the R project into its own folder within the repo? To contain the R project in a single folder and not pollute the dictionary repo.
* Can you add additional tests for the following validations:
* Test if a mandatory column is not present it throws an error (eg., fileID)
* Test if the value in a mandatory column is not present it throws an error (eg., fileID)
* Test if the value in a category column is not one of the categories, it throws an error (eg., destinations)
* Can you try to write each test so that its only testing one thing?
* For example, the warning with incorrect fileType is testing two things:
	* Its testing that an error is generated if a CSV file has a set as its part
	* Its testing that an error is generated if a set in an excel file is not present
* These will allow each test to be as small as possible, making it easy to debug why a test failed
* Can you try to write the test description in a human readable way? For example,
* standard run with no issues -> If the user passes in a dictionary with a valid files sheet, it should not throw an errors
* warning with incorrect filetype -> If the user passed in a dictionary where the part column in the files sheet is invalid, it should throw an error
* Should files-validation.R be renamed to create-release-files.R since that's the main function?

**Pull request closed without merging**